packages/langium/test/grammar/grammar-validator.test.ts

```diff
@@ -912,3 +912,128 @@ describe('Cross-reference to type union is only valid if all alternatives are AS
         expectNoIssues(validationResult);
     });
 });
+
+function detectEmptyRule(validationResult: ValidationResult, line1: number, col1: number, line2: number, col2: number) {
```

Comment:
Suggestion: With the suggestion above, instead of `line1, col1...` you can just pass the name of the rule as a string and use that to find the node that contains the error. That makes it easier to write and read all tests that use this function.

Reply:
Simplified tests to look for empty rules with a given name.